public/src/views/CalendarView.jsx

```diff
@@ -1,10 +1,108 @@
+  align-items: center;
+  margin: 1em auto;
+`;
+const CalendarFormContainer = styled.div`
+  ${({ theme }) => css`
+    width: ${theme.maxWidth};
+  `}
+`;
 
 export const CalendarView = () => {
+  const [day, setDayOrMonth] = useState(true);
+  const [data, setData] = useState([]);
+  const [shownDate, setShownDate] = useState(new Date());
+  const retrieveDate = async () => {
+    const date = formatDate(shownDate).substr(0, 10);
+    const response = await API.retrieveEventsForDay(date);
+    setData([response]);
+  };
+  const handleDateTimeInput = (e) => {
+    if (!e.target.validity.valid) return;
+    const [date, time] = e.target.value.split('T');
+    const [year, month, d] = date.split('-');
+    const [hour, minutes] = time.split(':');
+    const dt = new Date(Number(year), Number(month) - 1, Number(d), Number(hour), Number(minutes));
+    setShownDate(dt);
+  };
+  const retrieveWeek = async () => {
+    const END_OF_WEEK = 6;
+    const startOfWeek = shownDate.getDay() === 0
+      ? shownDate
+      : (() => { const a = new Date(); a.setDate(shownDate.getDate() - shownDate.getDay()); return a; })();
+    const endOfWeek = shownDate.getDay() === 6
+      ? shownDate
+      : (() => {
+        const a = new Date(); a.setDate(shownDate.getDate() + (END_OF_WEEK - shownDate.getDay())); return a;
+      })();
+    const response = await API.retrieveEventsForDuration(
+      formatDate(startOfWeek).substr(0, 10),
+      formatDate(endOfWeek).substr(0, 10),
+    );
+    const groupByDate = (ajaxData) => {
+      const groupedObj = ajaxData.reduce((acc, cur) => {
+        const [date, time] = cur.datetime.split('T');
+        Array.isArray(acc[date]) ? acc[date].push(cur) : acc[date] = [cur];
+        return acc;
+      }, {});
+      const groupedArr = [];
+      Object.keys(groupedObj).forEach((key) => {
+        groupedArr.push(groupedObj[key]);
+      });
+      return groupedArr;
+    };
+    const groupedData = groupByDate(response);
+    setData(groupedData);
+  };
+  useEffect(() => {
+    day ? retrieveDate() : retrieveWeek();
+  }, [day, shownDate]);
   return (
-    
-  )
+    <CalendarContainer>
+      <CalendarFormContainer>
+        <ComposableInput
+          label="Date and Time"
+          name="DateAndTime"
+          input={(
+            <StyledInput
+              id="DateAndTime"
+              type="datetime-local"
+              value={formatDate(shownDate)}
+              onChange={handleDateTimeInput}
+            />
+          )}
+        />
+        <ButtonGroup horizontal>
+          <CheckboxButton label="Day" name="Day" checked={day} onChange={() => setDayOrMonth(true)}>
+            Day
+          </CheckboxButton>
+          <CheckboxButton label="Week" name="Week" checked={!day} onChange={() => setDayOrMonth(false)}>
+            Week
+          </CheckboxButton>
+        </ButtonGroup>
+      </CalendarFormContainer>
+      <DaysContainer>
+        {/* eslint-disable-next-line */}
```

Comment:
Why?

Reply:
Cuz I needed to use index as part of (but not the entire) key for the list of components and Eslint was shitting a brick in fury over indexes being used as a key.